File: backend/PUGS/TravelPlanningService/Models/ChecklistItem.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TravelPlanningService.Models
{
    public class ChecklistItem
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public Guid TravelPlanId { get; set; }

        [ForeignKey(nameof(TravelPlanId))]
        public TravelPlan? TravelPlan { get; set; }

        [Required, MaxLength(150)]
        public string Title { get; set; } = string.Empty;

        public bool IsCompleted { get; set; } = false;
    }
}

File: frontend/src/models/ChecklistItem.js
// models/ChecklistItem.js
export class ChecklistItem {
  constructor({ id, travelPlanId, title, isCompleted }) {
    this.id = id;
    this.travelPlanId = travelPlanId;
    this.title = title;
    this.isCompleted = isCompleted;
  }
}